core/src/hir/methods/borrowing_param.rs

```diff
@@ -135,8 +135,55 @@ pub struct BorrowedLifetimeInfo<'tcx> {
 }
 
 impl<'tcx> BorrowingParamVisitor<'tcx> {
-    pub(crate) fn new(method: &'tcx Method, tcx: &'tcx TypeContext) -> Self {
-        let used_method_lifetimes = method.output.used_method_lifetimes();
+    pub(crate) fn new(
+        method: &'tcx Method,
+        tcx: &'tcx TypeContext,
+        force_include_slices: bool,
+    ) -> Self {
+        let mut used_method_lifetimes = method.output.used_method_lifetimes();
+
+        if force_include_slices {
+            if let Some(s) = &method.param_self {
+                if let hir::SelfType::Struct(s) = &s.ty {
+                    let st = s.resolve(tcx);
+                    for f in &st.fields {
+                        BorrowingParamVisitor::add_slices_to_used_lifetimes(
```

Comment:
thought: ugh, don't like having to recurse, but I guess that's inevitable.

Better long term design would be for the Lifetimes map to keep track of which lifetimes are tainted by slice-iness